mlflow/projects/_project_spec.py

```diff
@@ -0,0 +1,136 @@
+"""Internal utilities for parsing MLproject YAML files."""
+
+import os
+
+from six.moves import shlex_quote
+
+from mlflow import data
+
+class Project(object):
```

Comment:
For future reviewers, this is simply an extraction of the Project, EntryPoint, and Parameter classes from projects.py, with the addition of inlined imports and `_sanitize_param_dict` copied verbatim into a static method.